src/components/theme/__tests__/theme.test.js

```diff
@@ -0,0 +1,164 @@
+import merge from 'lodash/merge';
+import Color from 'color';
+import randomSwatches from '../../../../test/fixtures/theme/colors';
+import { theme } from '../theme';
+import { generateContrastColors, MIN_CONTRAST } from '../contrast-colors';
+import * as vars from '../../../styles/variables';
+
+describe('Validate generated color accessibility', () => {
+    const accessibleKeys = ['backgroundHover', 'backgroundActive', 'borderHover', 'borderActive'];
+    const contrastKey = 'foreground';
+
+    // Potentially problematic colors merged with random 1000 swatches from fixture set.
+    const colorCases = ['#ffffff', '#000000', '#232300', '#c50213', ...randomSwatches];
+
+    test.each(colorCases)('given %p as argument, returns accessible color contrast', colorKey => {
```

Comment:
This is going to add 1000+ tests to our runs. Are so many random swatches needed? Any impact on speed? Should this just be a single test with a loop inside it, instead?

Reply:
Yeah, this is a pretty gnarly one.. I think the solution that i'd like to do here is pull in a library like: https://www.npmjs.com/package/jest-expect-message so that I can get the proper information out of a failed expectation. The problem I encountered and why i'm using a jest each here is so that I know the failing swatch. Using jest-expect-message, I can output the swatch and also the particular accessible key check i'm doing. 

If possible, i'd like to do that one in a follow-up since it requires addition to the Jest configuration, unless you have some other thoughts around that one.

Reply:
overriding `expect()` might be problematic if they add a param of their own later. We may just want to write our own custom expect matcher instead

Reply:
A customer matcher could potentially work, but would still need a way to pipe in the metadata about the contrast ratios we are comparing.